### ISSUE TITLE
datetime.datetime.now() returns wrong time on Windows when env var TZ is set

### ISSUE THREAD

**Issue by jjYBdx4IL** [python/cpython#124286]
# Bug report

### Bug description:

Europe/Berlin is UTC+2 right now, not UTC+1

Works fine on Linux OSes (builds).

```cmd
C:\Users\work>python
Python 3.12.6 (tags/v3.12.6:a4a2d2b, Sep  6 2024, 20:11:23) [MSC v.1940 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.environ['TZ'] = 'Europe/Berlin'
>>> import datetime
>>> datetime.datetime.now()
datetime.datetime(2024, 9, 20, 12, 15, 24, 771320)
>>> datetime.datetime.now(tz=datetime.UTC)
datetime.datetime(2024, 9, 20, 11, 15, 45, 162005, tzinfo=datetime.timezone.utc)
>>>
```

### CPython versions tested on:

3.12

### Operating systems tested on:

Windows

**Comment by donBarbos** [comment]
Time zones really work strangely.
It's february now and in Berlin they use UTC+1 but i got next:
```python
>>> os.environ['TZ'] = 'Europe/Berlin'
>>> datetime.datetime.now() # my timzene UTC+4
datetime.datetime(2025, 2, 26, 2, 48, 1, 560149)
>>> datetime.datetime.now(tz=datetime.UTC) # current tz in Berlin: CET — Central European Time
datetime.datetime(2025, 2, 25, 22, 47, 59, 926276, tzinfo=datetime.timezone.utc) # should be 23 hourse
```
*tested on Linux and using `main` branch*

**Comment by ttw225** [comment]
# TL;DR

This issue is most likely caused by an outdated local timezone database, rather than a bug in CPython itself.

✅ Python officially recommends using the `tzdata` package as defined in [PEP 615](https://peps.python.org/pep-0615/)
to ensure accurate and up-to-date timezone data.

---

# Investigating Timezone Data Source in `zoneinfo`

## ZoneInfo Data Source Logic

The internal loading mechanism in CPython is as follows:
https://github.com/python/cpython/blob/main/Lib/zoneinfo/_zoneinfo.py#L61-L75
```python
@classmethod
def _new_instance(cls, key):
    obj = super().__new__(cls)
    obj._key = key
    obj._file_path = obj._find_tzfile(key)

    if obj._file_path is not None:
        file_obj = open(obj._file_path, "rb")
    else:
        file_obj = _common.load_tzdata(key)

    with file_obj as f:
        obj._load_file(f)

    return obj
```

Priority of data source:

1. Check system timezone database (`TZPATH`)
2. Fallback to `tzdata` package if not found

## Basic Test: Check Source and Offset

```python
import os
from datetime import datetime
from zoneinfo import ZoneInfo
from zoneinfo._tzpath import TZPATH
from typing import Optional

def get_local_tz_path(timezone: str) -> Optional[str]:
    for path in TZPATH:
        full_path = os.path.join(path, timezone)
        if os.path.exists(full_path):
            return full_path
    return None

def show_basic_timezone_info(timezone: str) -> None:
    try:
        tz = ZoneInfo(timezone)
        now = datetime.now(tz)
        offset = now.utcoffset()
        offset_hours = offset.total_seconds() / 3600 if offset else 0.0
        local_path = get_local_tz_path(timezone)
        source_info = local_path if local_path else "Not from local TZ database"

        print(f"Timezone: {timezone}")
        print(f"Current time: {now}")
        print(f"UTC offset: UTC{offset_hours:+.1f}")
        print(f"Source: {source_info}")
    except Exception as e:
        print(f"Error: {e}")
```

### Example usage

```python
>>> show_basic_timezone_info("Europe/Berlin")
Timezone: Europe/Berlin
Current time: 2025-06-15 05:30:19.564745+02:00
UTC offset: UTC+2.0
Source: /usr/share/zoneinfo/Europe/Berlin
```

---

## Force tzdata for Testing

Use the following to force tzdata usage instead of local TZPATH:

```python
import os
os.environ["PYTHONTZPATH"] = ""

from zoneinfo import ZoneInfo
from datetime import datetime

timezone = "Europe/Berlin"
tz = ZoneInfo(timezone)
now = datetime.now(tz)
offset = now.utcoffset()
offset_hours = offset.total_seconds() / 3600 if offset else 0.0

from zoneinfo._tzpath import TZPATH
def detect_timezone_source(name: str) -> str:
    for path in TZPATH:
        if os.path.exists(os.path.join(path, name)):
            return f"System TZ database: {os.path.join(path, name)}"
    try:
        import tzdata
        tzdata_path = os.path.join(os.path.dirname(tzdata.__file__), "zoneinfo", name)
        if os.path.exists(tzdata_path):
            return f"tzdata package: {tzdata_path}"
    except ImportError:
        pass
    return "Unknown source"

print(f"Timezone: {timezone}")
print(f"Current time: {now.isoformat()}")
print(f"UTC offset: UTC{offset_hours:+.1f}")
print(f"Source: {detect_timezone_source(timezone)}")
```

### Output:

```
Timezone: Europe/Berlin
Current time: 2025-06-15T10:25:25.895861+02:00
UTC offset: UTC+2.0
Source: tzdata package: /Users/user/.venv/lib/python3.13/site-packages/tzdata/zoneinfo/Europe/Berlin
```

## Summary

- `ZoneInfo` prioritizes system tzdata over `tzdata` package
- Ensuring your system tzdata is up to date, or explicitly using `tzdata`, helps avoid inconsistencies
